### PR TITLE
Make some buildtools py2/3 compatible

### PR DESCRIPTION
**Pull request by cowlicks**

### CODE REVIEW

File: utils/make-json.py [hunk 1]
```diff
@@ -57,13 +57,13 @@
...
     # FF version can find it.
-    xpath_ruleset(tree)[0].attrib["f"] = os.path.basename(fi).decode(encoding="UTF-8")
+    xpath_ruleset(tree)[0].attrib["f"] = os.path.basename(fi)
 
     for target in targets:
         # id is the current length of the rules list - i.e. the offset at which
         # this rule will be added in the list.
         json_output["targets"][target].append(len(json_output["rulesetStrings"]))
-    json_output["rulesetStrings"].append(etree.tostring(tree))
+    json_output["rulesetStrings"].append(etree.tounicode(tree))
```
Comment (cowlicks) [inline utils/make-json.py]: This file is where behavior changes would be if there are any.

**Comment by cowlicks** [comment]
It looks like this directly reverts some changes in #7136

Hmm I think the correct path forward is probably to adapt our python scripts to be 2/3 compatible, and update their shebangs as we do so.

The hardest change would be to `utils/zipfile_deterministic.py`, instead of updating that file it might be better to try to just use a linux utility. We could also just bundle it with `create_xpi.py` and use that directly as a script, since there are no external dependencies beyond that.

**Comment by koops76** [comment]
@cowlicks I don't think we need the builds to be deterministic since Tor Project makes their own builds.

**Comment by Bisaloo** [comment]
Related: https://github.com/EFForg/https-everywhere/issues/9993

**Comment by Hainish** [comment]
Superseded by more recent py3.6 rewrite.

**Comment by Hainish** [comment]
https://github.com/EFForg/https-everywhere/issues/6749

**Pull request closed without merging**